gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/user/register/RegisterEndpoint.java

```diff
@@ -40,30 +42,33 @@
  * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
  * @author GraviteeSource Team
  */
-public class RegisterEndpoint implements Handler<RoutingContext> {
+public class RegisterEndpoint extends AbstractEndpoint implements Handler<RoutingContext> {
 
     private static final Logger logger = LoggerFactory.getLogger(RegisterEndpoint.class);
 
     private final ThymeleafTemplateEngine engine;
+    private final Domain domain;
 
-    public RegisterEndpoint(ThymeleafTemplateEngine engine) {
+    public RegisterEndpoint(ThymeleafTemplateEngine engine, Domain domain) {
         this.engine = engine;
+        this.domain = domain;
     }
 
     @Override
     public void handle(RoutingContext routingContext) {
-        final HttpServerRequest request = routingContext.request();
-        final String error = request.getParam(ConstantKeys.ERROR_PARAM_KEY);
-        final String errorDescription = request.getParam(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY);
-        final String success = request.getParam(ConstantKeys.SUCCESS_PARAM_KEY);
-        final String warning = request.getParam(ConstantKeys.WARNING_PARAM_KEY);
-        final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
-        final String clientId = request.getParam(Parameters.CLIENT_ID);
+        HttpServerRequest request = routingContext.request();
+        copyValue(request, routingContext, ConstantKeys.SUCCESS_PARAM_KEY);
+        copyValue(request, routingContext, ConstantKeys.WARNING_PARAM_KEY);
+        Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
+        routingContext.put(ConstantKeys.PASSWORD_CONFIG_PARAM_KEY, PasswordSettings.getInstance(client, domain));
```

Comment:
Do not inject the Optional here but rather the value, this way we keep coherency with the others attributes.

Reply:
ok